modules/core/src/smithy4s/capability/Transformation.scala

```diff
@@ -0,0 +1,37 @@
+package smithy4s
+package capability
+
+import kinds._
+
+/**
+  * Heterogenous function construct, allows to abstract over various kinds of functions
+  * whilst providing an homogenous user experience without the user having to
+  * manually lift functions from one kind to the other.
+  *
+  * Used to reduce the noise of transformations
+  */
+trait Transformation[Func, Input, Output] {
```

Comment:
This is probably the most complicated thing here : essentially this is a construct that says "I, Func, am a polymorphic function of some kind,  and when given an input, I can produce an output". For example, we use it to express the fact that we know how to apply a good old `F ~> G` (with `F[_] ` and `G[_]`) on something that looks like `Foo[F[_, _, _, _, _]]`. 

This allows to make it **simpler** for the user to apply generic transformations on the instances of the generated interfaces, without having to worry about the kinds at hand. 

TBH : I'm not fully certain it's a good idea. I just want to avoid having the users having to worry about the humongous kinds we deal with in this library, and for cat's FunctionK to just "work" with it modulo a simple import.